shell/browser/api/event.cc

```diff
@@ -21,11 +21,15 @@ Event::Event() {}
 Event::~Event() {
   if (callback_) {
     v8::Isolate* isolate = electron::JavascriptEnvironment::GetIsolate();
-    v8::HandleScope scope(isolate);
-    auto message = gin::DataObjectBuilder(isolate)
-                       .Set("error", "reply was never sent")
-                       .Build();
-    SendReply(isolate, message);
+    // If there's no current context, it means we're shutting down, so we don't
+    // need to send an event.
+    if (!isolate->GetCurrentContext().IsEmpty()) {
+      v8::HandleScope scope(isolate);
+      auto message = gin::DataObjectBuilder(isolate)
```

Comment:
Could we more specifically check this? e.g with `Browser::Get()->is_shutting_down()`?

Reply:
That would be a less specific check IMO. The crash is that the current context is empty. This check directly corresponds to the condition that matters here.

The current context is cleared partway through shutting down (https://github.com/electron/electron/blob/023c89265ffa8bdb21fac89474e1441822202f39/shell/browser/javascript_environment.cc#L88). This will work before then, and not afterwards.